commands/system/help.js

```diff
@@ -16,67 +17,98 @@ class Help extends Command {
+
+        embed
+          .setTitle(title)
+          .setDescription([description, output.length > perpage ? tip : ""])
+          .setArray(output)
+          .formatField("Categories", el => el);
+
+        if (isPage) embed.setPage(parseInt(type));
+      } else {
+        let n = 0;
+        sorted.forEach(c => {
+          if (c.help.category.toLowerCase() === type.toLowerCase()) {
+            n++;
+          }
+        });
+      
+        const output = [];
+        let num = 0;
+        const pg = parseInt(page) && parseInt(page) <= Math.ceil(n / perpage) ? parseInt(page) : 1;
+        for (const c of sorted.values()) {
+          if (c.help.category.toLowerCase() === type.toLowerCase()) {
+            if (c.help.category === "Owner" && level < 10 ) return;
+            if (c.help.category === "NSFW" && !message.channel.nsfw) return;
             if (level < this.client.levelCache[c.conf.permLevel]) return;
-            output += `\n\`${message.settings.prefix + c.help.name}\` | ${c.help.description.length > 50 ? c.help.description.slice(0,50) +"...": c.help.description}`;
+            output.push(`\`${message.settings.prefix + c.help.name}\` | ${c.help.description.length > 50 ? c.help.description.slice(0,50) +"...": c.help.description}`);
+            num++;
           }
-          num++;
+        }
+      
+        if (num) {
+          embed
+            .setTitle("Command category help")
+            .setDescription(`A list of commands in the ${type} category.\n(Total of ${num} commands in this category)\n\nTo get help on a specific command do \`${message.settings.prefix}help <command>\`${num > perpage ? `\n\nTo skip to a page for more commands:\n\tReact with ↗; or\n\tDo \`${message.settings.prefix}help <category> <page-num>\`` : "" }`)
+            .setArray(output)
+            .setPage(pg)
+            .formatField("Commands", el => el);
         }
       }
-    
-      if (num) {
-        embed.setTitle("Command category help")
-          .setDescription(`A list of commands in the ${type} category.\n(Total of ${num} commands in this category)\n\nTo get help on a specific command do \`${message.settings.prefix}help <command>\`\n\n${num > 10 && pg === 1 ? `To view more commands do\` ${message.settings.prefix}help <category> 2\`` : "" }`)
-          .addField("Commands", output);
-      }
-    }
 
-    if (this.client.commands.has(type) || this.client.commands.some(command => command.conf.aliases.includes(type))) {
-      const cm = this.client.commands.get(type) || this.client.commands.get(this.client.aliases.get(type));
-      if (level < this.client.levelCache[cm.conf.permLevel]) return;
-      embed.setTitle(cm.help.name.toProperCase())
-        .addField("Command description", cm.help.description)
-        .addField("Command usage", `\`${cm.help.usage}\``)
-        .addField("Command aliases", cm.conf.aliases.length == 0 ? "None" : cm.conf.aliases.join(", ") );
+      return await embed.build();
```

Comment:
return await is kinda useless since if you return out of a async await it will auto await Promises anyways

Reply:
Correction: `return await` is kinda useless as [AsyncFunctions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/AsyncFunction)s return a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise). In order to get the result, you'd await that promise, and if you returned another promise, it'd await that one aswell.